state-synchronizer/src/coordinator.rs

```diff
@@ -461,6 +461,20 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
         let local_epoch = self.local_state.epoch();
         counters::set_version(counters::VersionType::Synced, synced_version);
         counters::set_version(counters::VersionType::Committed, committed_version);
+        counters::set_timestamp(
+            counters::TimestampType::Synced,
+            self.executor_proxy
+                .get_vm_timestamp_for_version(synced_version)?,
```

Comment:
This calls into storage to get the timestamp for a version? I guess you probably don't want to do it (especially multiple times in the critical path)?

Reply:
I don't this it is multiple times per chunk. Thoughts on alternative approaches?

Reply:
Service thread for logging? 😝😝😝 But probably log it even more less frequently. For example, log commited timestamp only when a ledger_info is written, and when it's the case, the synced timestamp should be the same?

Reply:
K. I was exaggerating. Probably fine doing it twice for each chunk.